external/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -201,4 +201,25 @@ private[sql] object AvroUtils extends Logging {
       }
     }
   }
+
+  /**
+   * Extract a single field from `avroFields` which has the same name has `catalystField`,
+   * performing the matching with proper case sensitivity according to [[SQLConf.CASE_SENSITIVE]].
+   *
+   * @param avroFields The fields within which to search for a match.
+   * @param catalystField The catalyst field for which to search for a match.
+   * @return `Some(match)` if a matching Avro field is found, otherwise `None`.
+   */
+  private[avro] def getAvroFieldForCatalystField(
```

Comment:
This has nothing to do with catalyst field, only the name, how about
```
def getAvroFieldByName(avroSchema: Schema, name: String): Option[Schema.Field] = {
  val isEqual = SQLConf.get.resolver
  avroSchema.getFields.asScala.filter ...
}
```

Reply:
+1

Reply:
Completely right, good call. Also thanks for the pointer on `SQLConf.resolver`, that is much nicer!